From 0995ca8ee2b7cbd768921b0a4e2f88a98af09c92 Mon Sep 17 00:00:00 2001
From: Mikau <mikau@aaathats3as.com>
Date: Mon, 17 Apr 2023 16:37:10 +0200
Subject: [PATCH] Handle "Not Charging" state in battery indicator

When the battery has reached the target charge level, some devices
use the status "Not Charging". LXpanel should not interpret this to mean
"Discharging", but rather the same as "Full".

This is also likely to fix github issue #50, reported by embedeo.

(commit message amended by committer)
---
 plugins/batt/batt_sys.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/plugins/batt/batt_sys.c b/plugins/batt/batt_sys.c
index 5a40273e..eb807a3c 100644
--- a/plugins/batt/batt_sys.c
+++ b/plugins/batt/batt_sys.c
@@ -396,6 +396,7 @@ gboolean battery_is_charging( battery *b )
     return ( strcasecmp( b->state, "Unknown" ) == 0
             || strcasecmp( b->state, "Full" ) == 0
             || strcasecmp( b->state, "Charging" ) == 0
+            || strcasecmp( b->state, "Not Charging" ) == 0
             || b->current_now == 0 ); /* bug sf.net, #720 */
 }
 
